alerting/notifier/channels/kafka.go

```diff
@@ -48,12 +54,36 @@ type KafkaNotifier struct {
 }
 
 type kafkaSettings struct {
-	Endpoint    string `json:"kafkaRestProxy,omitempty" yaml:"kafkaRestProxy,omitempty"`
-	Topic       string `json:"kafkaTopic,omitempty" yaml:"kafkaTopic,omitempty"`
-	Description string `json:"description,omitempty" yaml:"description,omitempty"`
-	Details     string `json:"details,omitempty" yaml:"details,omitempty"`
+	Endpoint       string `json:"kafkaRestProxy,omitempty" yaml:"kafkaRestProxy,omitempty"`
+	Topic          string `json:"kafkaTopic,omitempty" yaml:"kafkaTopic,omitempty"`
+	Description    string `json:"description,omitempty" yaml:"description,omitempty"`
+	Details        string `json:"details,omitempty" yaml:"details,omitempty"`
+	BasicAuth      bool   `json:"basicAuth,omitempty" yaml:"basicAuth,omitempty"`
+	BasicAuthUser  string `json:"basicAuthUser,omitempty" yaml:"user,omitempty"`
+	PasswordSource string `json:"passwordSource,omitempty" yaml:"passwordSource,omitempty"`
+	PassFilePath   string `json:"passwordFilePath,omitempty" yaml:"passwordFilePath,omitempty"`
```

Comment:
I do not think that we are ready to support this in Grafana notifiers, at least in the current implementation.
1. I think this creates quite a serious security vulnerability because there is nothing to stop a malicious user that has access to API to specify a path to SQLlite database, or grafana config file and send it in the Authz header back to malicious API endpoint.
2. None of Grafana notifiers support loading secrets from file. This might change when we start replacing Grafana implementation of notifiers with the Alertmanager ones, and we will have to figure out how to make it secure.

cc @gotjosh what do you think?

Reply:
Agreed with Yuri here, this does not belong within Grafana. We already have a way in Mimir to unsupport these fields in Cloud. 

So when we move over, we'll get it for free.

Reply:
Got it! That's a valid concern.
I would also like to point out that it is current supported to supply Grafana's DB password from a file here: https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#file-provider

Would it help if I limit the size of the password on the file? To say 32 characters? 

The reasons I wanted to support this is so that when Grafana is deployed in a kubernetes environment, I would prefer that the password is stores in a kubernetes secret and mounted onto the grafana container. This also allows me to rotate the password frequently without restarting grafana each time.

Reply:
The difference between these two cases is that Grafana config can only be changed via updating a config file or via environment variable, and that requires access to the host. In the case of notification policies, this is a purely API call, which has much more relaxed constraints. Basically, any user with `Editor` role is able to create a contact point.

> This also allows me to rotate the password frequently without restarting grafana each time.

You can use an API call to rotate secrets instead. It can be a sidecar container that would do that.

Reply:
Would it help if I limit the size of the password on the file? To say 32 characters? Or would you guys prefer that I remove this altogether?

Reply:
Thanks @yuri-tceretian 
I'll remove this part and leave you guys an update. Thanks for the review!

Reply:
One way I can recommend: introduce support for kafka in the alertmanager repository https://github.com/prometheus/alertmanager where you can introduce support for loading secrets from files. Then if it is accepted, you can configure Grafana to route all alerts to external Alertmanager instead of the internal one.

Reply:
Ah, that makes sense! Thanks @yuri-tceretian! I'll see if I can find the time to work on that.

Reply:
Have remove the reading from file part. Will mark this resolved. I will get to the alertmanager changes in the future.